app/views/puppetclasses/_class_selection.html.erb

```diff
@@ -1,6 +1,6 @@
   <% if obj.errors[:puppetclasses].any? %>
     <%= alert :class => "alert-block alert-danger base in fade",
```

Comment:
if you add `has-error` to the alert class and remove `.form-group` from app/assets/javascript/application.js:62, the focus will work using the generic error focus handling.